nixos/modules/services/system/userborn.nix

```diff
@@ -83,8 +103,20 @@ in
         message = "system.activationScripts.users has to be empty to use userborn";
       }
       {
-        assertion = immutableEtc -> (cfg.passwordFilesLocation != "/etc");
-        message = "When `system.etc.overlay.mutable = false`, `services.userborn.passwordFilesLocation` cannot be set to `/etc`";
+        assertion = (immutableEtc && !cfg.static) -> (cfg.passwordFilesLocation != "/etc");
+        message = "When `system.etc.overlay.mutable = false` and `services.userborn.static = false`, `services.userborn.passwordFilesLocation` cannot be set to `/etc`";
+      }
+      {
+        assertion = cfg.static -> cfg.passwordFilesLocation == "/etc";
+        message = "When `services.userborn.static = true`, `services.userborn.passwordFilesLocation` must be set to `/etc`";
+      }
```

Comment:
I would get rid of this assertion. I doesn't stop any nefarious/dangerous use case. Too strict assertions usually bite us later.